apps/prairielearn/src/lib/authn.ts

```diff
@@ -13,34 +14,35 @@ import { HttpRedirect } from './redirect.js';
 
 const sql = sqldb.loadSqlEquiv(import.meta.url);
 
-/**
- * @typedef {Object} LoadUserOptions
- * @property {boolean} [redirect] - Redirect after processing?
- */
-/**
- * @typedef {Object} LoadUserAuth
- * @property {string} [uid]
- * @property {string | null} [uin]
- * @property {string | null} [name]
- * @property {string | null} [email]
- * @property {string} [provider]
- * @property {number} [user_id] - If present, skip the users_select_or_insert call
- * @property {number | string | null} [institution_id]
- */
-/**
- * @param {import('express').Request} req
- * @param {import('express').Response} res
- * @param {LoadUserAuth} authnParams
- * @param {LoadUserOptions} [optionsParams]
- */
-export async function loadUser(req, res, authnParams, optionsParams = {}) {
-  let options = { redirect: false, ...optionsParams };
-
-  let user_id;
-  if ('user_id' in authnParams) {
+interface LoadUserOptions {
+  /** Redirect after processing? */
+  redirect?: boolean;
+}
+
+export interface LoadUserAuth {
+  uid?: string;
+  uin?: string | null;
+  name?: string | null;
+  email?: string | null;
+  provider: string;
+  /** If present, skip the users_select_or_insert call */
+  user_id?: number;
```

Comment:
Newer code treats IDs coming out of the DB as strings; we can do both for backwards compatibility, though in practice these are almost always just passed directly to Postgres which doesn't really care.

```suggestion
  user_id?: number | string;
```